src/main/java/dao/FileReaderImpl.java

```diff
@@ -0,0 +1,34 @@
+package dao;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import model.FruitTransaction;
+import model.OperationType;
+
+public class FileReaderImpl implements dao.FileReader {
+
+    @Override
+    public List<FruitTransaction> readFile(String fileName) {
```

Comment:
The method signature 'readFile' suggests that it reads a generic file, but the implementation is specific to reading 'FruitTransaction' data. Consider renaming the method to reflect its specific purpose, like 'readTransactions' or 'readFruitTransactions'.